tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -348,7 +349,7 @@ def update_leader(self):
         while retries > 0:
             for p in self.processors:
                 found = list(p.node.account.ssh_capture("grep \"Finished assignment for group\" %s" % p.LOG_FILE, allow_fail=True))
-                if len(found) == self.leader_counter[p] + 1:
+                if len(found) >= self.leader_counter[p] + 1:
```

Comment:
This is one fix. If rebalance happens quickly, we might see this message for the second rebalance early. Cf. #6763

Reply:
After https://github.com/apache/kafka/pull/6779 is merged, I can rebase this PR -- we still want this fix I guess :)